app/schemas/webhooks/zalo_schema.py

```
@@ -0,0 +1,108 @@
+class ZaloUserSendLocationEvent(ZaloBaseEvent):
+    event_name: Literal["user_send_location"]
+    sender: ZaloSender
+    recipient: ZaloRecipient
+    message: ZaloLocationMessage
+
+
+# ---------------------------------------------------------------------------
+# Union type + discriminated parsing
+# ---------------------------------------------------------------------------
+ZaloWebhookPayload = Annotated[
+    Union[ZaloUserSendTextEvent, ZaloUserSendImageEvent, ZaloUserSendLocationEvent],
```

Ruff (0.15.11)

[warning] Use `X | Y` for type annotations

Convert to `X | Y`

(UP007)